src/phoenix/experiments/types.py

```diff
@@ -427,20 +512,101 @@ def url(self) -> str:
     def info(self) -> str:
         return f"🔗 View this experiment: {self.url}"
 
-    def add(self, eval_summary: EvaluationSummary) -> "RanExperiment":
-        eval_summaries = (eval_summary, *self.eval_summaries)
+    def __post_init__(self) -> None:
+        runs = {
+            id_: (
+                _ExperimentRunWithExample(run, example)
+                if (example := self.dataset.examples.get(run.dataset_example_id))
+                else run
```

Comment:
When would a run not have an associated experiment?

Reply:
Probably never, but there's no guarantee, and i didn't want to raise an exception here since this is all just for the `__repr__`

Reply:
I.e., if we expect the example to be there, can we just do `self.dataset.example[run.dataset_example_id]`?

Reply:
it's probably more complicated than warranted on the server side in terms of the payloads for sending and fetching experiments